lib/miq_automation_engine/service_models/miq_ae_service_manageiq-providers-vmware-infra_manager-vm.rb

```diff
@@ -15,5 +15,10 @@ def add_disk(disk_name, disk_size_mb, options = {})
     def remove_disk(disk_name, options = {})
       sync_or_async_ems_operation(options[:sync], "remove_disk", [disk_name, options])
     end
+
+    def move_into_folder(folder)
+      raise ArgumentError, "folder must be a MiqAeServiceEmsFolder" unless folder.kind_of?(MiqAeMethodService::MiqAeServiceEmsFolder)
+      ar_method { @object.move_into_folder(EmsFolder.find_by(:id => folder.id)) }
```

Comment:
The call to VMware is here and is synchronous so we should wrap this in the same `sync_or_async_ems_operation` method as the above methods which will require the (initially added, then removed) `options` hash parameter.

See `moveIntoFolder` method: https://github.com/ManageIQ/vmware_web_service/blob/501b80924d89573aed8782aa59a4f6b75624a64c/lib/VMwareWebService/MiqVimFolder.rb#L84